File: lib/spiral/lib_wasm.php
<?php
namespace Lib;

require_once(__ROOT__.'/../../../../../lib/spiral/file_system.fs.phpx');
require_once(__ROOT__.'/../../../../../lib/spiral/sm.fs.phpx');
require_once(__ROOT__.'/../../../../../lib/spiral/trace.fs.phpx');

#0
function SpiralTrace_get_trace_state($unitVar) {
    return \trace\get_trace_state(NULL);
}

#1
function SpiralTrace_trace($x) {
    return \trace\trace($x);
}

#2
$GLOBALS['SpiralTrace_US0_0'] = new \Trace\US0_US0_0();

#3
$GLOBALS['SpiralTrace_US0_1'] = new \Trace\US0_US0_1();

#4
$GLOBALS['SpiralTrace_US0_2'] = new \Trace\US0_US0_2();

#5
$GLOBALS['SpiralTrace_US0_3'] = new \Trace\US0_US0_3();

#6
$GLOBALS['SpiralTrace_US0_4'] = new \Trace\US0_US0_4();

#7
function SpiralSm_concat($x) {
    return \sm\concat($x);
}

#8
function SpiralSm_contains($x) {
    return \sm\contains($x);
}

#9
function SpiralSm_ellipsis($x) {
    return \sm\ellipsis($x);
}

#10
function SpiralSm_ellipsis_end($x) {
    return \sm\ellipsis_end($x);
}

#11
function SpiralSm_ends_with($x) {
    return \sm\ends_with($x);
}

#12
function SpiralSm_format_exception($x) {
    return \sm\format_exception($x);
}

#13
function SpiralSm_replace($x) {
    return \sm\replace($x);
}

#14
function SpiralSm_slice($x) {
    return \sm\slice($x);
}

#15
function SpiralSm_split($x) {
    return \sm\split($x);
}

#16
function SpiralSm_split_string($x) {
    return \sm\split_string($x);
}

#17
function SpiralSm_starts_with($x) {
    return \sm\starts_with($x);
}

#18
function SpiralSm_substring($x) {
    return \sm\substring($x);
}

#19
function SpiralSm_to_lower($x) {
    return \sm\to_lower($x);
}

#20
function SpiralSm_trim($x) {
    return \sm\trim($x);
}

#21
function SpiralSm_trim_start($x) {
    return \sm\trim_start($x);
}

#22
function SpiralSm_trim_end($x) {
    return \sm\trim_end($x);
}

#23
function SpiralFileSystem_get_source_directory($unitVar) {
    return \file_system\get_source_directory(NULL);
}

#24
function SpiralFileSystem_find_parent($x) {
    return \file_system\find_parent($x);
}

#25
function SpiralFileSystem_create_temp_directory($unitVar) {
    return \file_system\create_temp_directory(NULL);
}

#26
function SpiralFileSystem_create_temp_directory_name($unitVar) {
    return \file_system\create_temp_directory_name(NULL);
}

File: lib/spiral/lib_wasm.dart
// ignore_for_file: camel_case_types, constant_identifier_names, non_constant_identifier_names, unnecessary_this
import '../../fable_modules/fable_library/String.dart' as string;
import '../../fable_modules/fable_library/Types.dart' as types;
import '../../../../../../../../lib/spiral/file_system.fsx' as file_system;
import '../../../../../../../../lib/spiral/networking.fsx' as networking;
import '../../../../../../../../lib/spiral/runtime.fsx' as runtime;
import '../../../../../../../../lib/spiral/sm.fsx' as sm;
import '../../../../../../../../lib/spiral/trace.fsx' as trace;

void Function(String Function()) Function(String Function()) SpiralTrace_trace(trace.US0 x) => trace.trace(x);

String Function(Iterable<String>) SpiralSm_concat(String x) => sm.concat(x);

bool Function(String) SpiralSm_contains(String x) => sm.contains(x);

String Function(String) SpiralSm_ellipsis(int x) => sm.ellipsis(x);

String Function(String) SpiralSm_ellipsis_end(int x) => sm.ellipsis_end(x);

bool Function(String) SpiralSm_ends_with(String x) => sm.ends_with(x);

String SpiralSm_format_exception(dynamic x) => sm.format_exception(x);

String Function(String) Function(String) SpiralSm_replace(String x) => sm.replace(x);

String Function(String) Function(int) SpiralSm_slice(int x) => sm.slice(x);

List<String> Function(String) SpiralSm_split(String x) => sm.split(x);

List<String> Function(String) SpiralSm_split_string(List<String> x) => sm.split_string(x);

bool Function(String) SpiralSm_starts_with(String x) => sm.starts_with(x);

String Function(String) Function(int) SpiralSm_substring(int x) => sm.substring(x);

String SpiralSm_to_lower(String x) => sm.to_lower(x);

String SpiralSm_trim(String x) => sm.trim(x);

String Function(String) SpiralSm_trim_start(List<int> x) => sm.trim_start(x);

String Function(String) SpiralSm_trim_end(List<int> x) => sm.trim_end(x);

String SpiralFileSystem_get_repository_root() => file_system.get_repository_root();

String SpiralFileSystem_get_source_directory() => file_system.get_source_directory();

String Function(String) Function(bool) SpiralFileSystem_find_parent(String x) => file_system.find_parent(x);

types.Tuple2<String, types.IDisposable> SpiralFileSystem_create_temp_directory() => file_system.create_temp_directory();

String SpiralFileSystem_create_temp_directory_name() => file_system.create_temp_directory_name();

void set_trace_level(trace.US0 new_level) {
    if (file_system.State_trace_state == null) {
        string.toConsole(string.printf('@1'));
    } else {
        switch (new_level.tag) {
            case 1:
                types.value(file_system.State_trace_state).item4.l0 = const file_system.US0(/* US0_1 */ 1);
                break;
            case 2:
                types.value(file_system.State_trace_state).item4.l0 = const file_system.US0(/* US0_2 */ 2);
                break;
            case 3:
                types.value(file_system.State_trace_state).item4.l0 = const file_system.US0(/* US0_3 */ 3);
                break;
            case 4:
                types.value(file_system.State_trace_state).item4.l0 = const file_system.US0(/* US0_4 */ 4);
                break;
            default:
                types.value(file_system.State_trace_state).item4.l0 = const file_system.US0(/* US0_0 */ 0);
        }
    }
    if (networking.State_trace_state == null) {
        string.toConsole(string.printf('@2'));
    } else {
        switch (new_level.tag) {
            case 1:
                types.value(networking.State_trace_state).item4.l0 = const networking.US0(/* US0_1 */ 1);
                break;
            case 2:
                types.value(networking.State_trace_state).item4.l0 = const networking.US0(/* US0_2 */ 2);
                break;
            case 3:
                types.value(networking.State_trace_state).item4.l0 = const networking.US0(/* US0_3 */ 3);
                break;
            case 4:
                types.value(networking.State_trace_state).item4.l0 = const networking.US0(/* US0_4 */ 4);
                break;
            default:
                types.value(networking.State_trace_state).item4.l0 = const networking.US0(/* US0_0 */ 0);
        }
    }
    if (runtime.State_trace_state == null) {
        string.toConsole(string.printf('@3'));
    } else {
        switch (new_level.tag) {
            case 1:
                types.value(runtime.State_trace_state).item4.l0 = const runtime.US0(/* US0_1 */ 1);
                break;
            case 2:
                types.value(runtime.State_trace_state).item4.l0 = const runtime.US0(/* US0_2 */ 2);
                break;
            case 3:
                types.value(runtime.State_trace_state).item4.l0 = const runtime.US0(/* US0_3 */ 3);
                break;
            case 4:
                types.value(runtime.State_trace_state).item4.l0 = const runtime.US0(/* US0_4 */ 4);
                break;
            default:
                types.value(runtime.State_trace_state).item4.l0 = const runtime.US0(/* US0_0 */ 0);
        }
    }
    if (trace.State_trace_state == null) {
        string.toConsole(string.printf('@4'));
    } else {
        types.value(trace.State_trace_state).item4.l0 = new_level;
    }
}

trace.US0 get_trace_level() {
    if (trace.State_trace_state == null) {
        string.toConsole(string.printf('@6'));
        return const trace.US0(/* US0_0 */ 0);
    } else {
        return types.value(trace.State_trace_state).item4.l0;
    }
}

